polyply/src/gen_coords.py

```diff
@@ -236,6 +236,22 @@ def gen_coords(toppath,
         LOGGER.info("loading grid",  type="step")
         grid = np.loadtxt(grid)
 
+    # where to get the box size from
+    if box is not None and not np.array_equal(topology.box, box):
+        msg = ("A box is provided via the -box command line "
+               "and the starting coordinates. We consider the "
+               "the box of starting coordinates as correct. ")
+        LOGGER.warning(msg,  type="warning")
+        box = topology.box
```

Comment:
If the box of input coordinates is not taken, it will always lead to incorrect pbc completion. This behavior is actually a big problem leading to this PR. So yes the CLI is easier to change but it can never overrule the input coordinates. At best it can be consistent.

Reply:
Ah, I didn't consider the effect the box would have on PBC unwrapping the coordinates. Agreed then :)